contracts/tips/src/msg.rs

```diff
@@ -0,0 +1,282 @@
+use crate::error::ContractError;
+use crate::state::StateServiceFee;
+use cosmwasm_std::{Addr, Coin, Uint128, Uint64};
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+
+/// Fees required to execute [`ExecuteMsg::SendTip`].
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+#[serde(rename_all = "snake_case")]
+pub enum ServiceFee {
+    /// Represents a fixed fee that is deducted from the tip.
+    Fixed {
+        /// Coins that are deducted from the tip.
+        amount: Vec<Coin>,
+    },
+    /// Represents a percentage that is deducted from the tip.
+    Percentage {
+        /// Percentage value.
+        value: Uint128,
+        /// Percentage decimals.
+        /// Example if this value is 3 means that the value should have 3 decimals numbers so
+        /// if value is
+        /// - 1 means 0,001%
+        /// - 10 means 0,01%
+        /// - 100 means 0,1%
+        /// - 1000 means 1%
+        /// - 10000 means 10%
+        /// - 100000 means 100%
+        decimals: u32,
```

Comment:
This is extremely confusing to read and understand. I think the overall explanation can be improved somehow to make it more clear